### ISSUE TITLE
TrainSpec.learning_rate is silently ignored by the lerobot_local trainer (every other backend honors it)

### ISSUE THREAD

**Issue by mertkrgl** [strands-labs/robots#1020]
## Summary

`TrainSpec` exposes `learning_rate`, the `train_policy` tool forwards it as a first-class parameter, and four of the five training backends honor it — but the `lerobot_local` trainer never wires it into the config it builds, so the policy's training-preset LR is always used. A user who sets `learning_rate` is silently misled.

## Steps to reproduce

    from strands_robots.training import TrainSpec, create_trainer
    spec = TrainSpec(dataset_root=..., output_dir=..., learning_rate=5e-5,
                     extra={"policy_type": "act"})
    cfg = create_trainer("lerobot_local").build_config(spec)
    print(cfg.policy.optimizer_lr)   # -> 1e-05 (the ACT preset), not 5e-05

Run today on main: requested `5e-05` → `cfg.policy.optimizer_lr = 1e-05`, and `build_command()` emits no `--policy.optimizer_lr` flag either.

## Actual behavior

Confirmed on a real GPU run (two-phase ACT training, base + post-tune, Colab A100): both phases logged `lr:1.0e-05` (the ACT preset) regardless of the `TrainSpec.learning_rate` value.

## Root cause / asymmetry

- `training/lerobot.py` contains no reference to `learning_rate` — neither `build_config()` (typed path) nor `build_command()` (argv-parity path) maps it.
- Every other backend uses the field: PPO (`rl/ppo.py`) and FastSAC (`rl/fast_sac.py`) bind it to their Adam optimizers; GR00T forwards `--learning_rate`; Cosmos forwards `optimizer.lr`.
- The agent-facing `train_policy` tool exposes `learning_rate` as a parameter and passes it into the spec — so the agent path advertises a knob the lerobot backend ignores.
- A working escape hatch exists and is even exercised by the test suite (`extra={"policy.optimizer_lr": ...}`), which makes the ignored first-class field the odd one out.
- In fact `training/lerobot.py` reads **every other applicable `TrainSpec` field** (dataset_root / dataset_repo_id / streaming / base_model / output_dir / steps / global_batch_size / save_freq / num_gpus / num_nodes / resume / seed / method / lora_* / tune / val_episodes) — `learning_rate` is the only universal field with zero references in the file.

## Why this isn't the documented tolerance rule

`TrainSpec`'s docstring does let backends "read the fields they support and ignore the rest" — but fields with designed partial support say so on the field itself (`seed`: "best-effort; not all backends expose it"; `streaming`: "LeRobot-only"; `tune`: "Ignored by backends that don't" expose the toggles). `learning_rate` carries no such caveat, sits in the universal block, and is advertised by the `train_policy` tool — this reads as an accidental gap, not designed partial support. (A field-by-backend sweep found sibling gaps only in the Cosmos3 trainer — filing those separately.)

## Expected behavior

An explicit `learning_rate` reaches lerobot's optimizer config (`policy.optimizer_lr`, which the policy's `get_optimizer_preset()` reads) — or is rejected loudly. Silently ignoring it is the one behavior that can't be right; same class as #897 / #968 / #979 (make silent failure loud).

One design note: the field currently defaults to `1e-4`, so wiring it unconditionally would change every *default* lerobot run (the ACT preset is 1e-5). The natural shape is the `seed` pattern the spec already uses: default `None` = "the backend's own default", an explicit value = honored everywhere.

## Environment

main @eb48cdd · lerobot 0.5.1 · confirmed on Colab A100 (real run) and locally (config-level, Python 3.12)
